hack/image_sync/sync-images.sh

```diff
@@ -0,0 +1,779 @@
+        fi
+    done
+    
+    if [[ $update_count -gt 0 ]]; then
+        print_success "Updated sample files: $old_tag → $new_tag ($update_count files updated)"
+    elif [[ $error_count -gt 0 ]]; then
+        print_warning "Some sample files could not be updated ($error_count files had issues)"
+    else
+        print_info "No sample files needed updating for $old_tag → $new_tag"
+    fi
+    
+    # Return success even if some files couldn't be updated - this is not a critical failure
+    return 0
+}
+
+# Function to extract image tags from Istio upstream samples
+extract_istio_tags() {
+    local temp_dir
+    temp_dir=$(mktemp -d)
+    local istio_sample_urls=(
+        "https://raw.githubusercontent.com/istio/istio/master/samples/helloworld/helloworld.yaml"
+        "https://raw.githubusercontent.com/istio/istio/master/samples/httpbin/httpbin.yaml"
+        "https://raw.githubusercontent.com/istio/istio/master/samples/tcp-echo/tcp-echo-ipv4.yaml"
+    )
+    
+    print_info "Fetching Istio upstream samples to extract official tags..."
+    
+    declare -A istio_tags
+    
+    for url in "${istio_sample_urls[@]}"; do
+        local filename
+        filename=$(basename "$url")
+        local filepath="$temp_dir/$filename"
+        
+        print_info "  Fetching $filename"
+        if curl -sSfL "$url" -o "$filepath"; then
+            # Extract image references and their tags
+            while IFS= read -r line; do
+                if [[ "$line" =~ image:[[:space:]]*docker\.io/istio/([^:]+):([^[:space:]]+) ]]; then
+                    local image_name="${BASH_REMATCH[1]}"
+                    local tag="${BASH_REMATCH[2]}"
+                    istio_tags["$image_name"]="$tag"
+                    print_info "    Found: docker.io/istio/$image_name:$tag"
+                elif [[ "$line" =~ image:[[:space:]]*([^:]+/go-httpbin):([^[:space:]]+) ]]; then
+                    local image_base="${BASH_REMATCH[1]}"
+                    local tag="${BASH_REMATCH[2]}"
+                    istio_tags["httpbin"]="$tag"
+                    print_info "    Found: $image_base:$tag"
+                fi
+            done < "$filepath"
+        else
+            print_warning "  Failed to fetch $filename"
+        fi
+    done
+    
+    rm -rf "$temp_dir"
+    
+    # Store results in global array
+    for key in "${!istio_tags[@]}"; do
+        echo "$key=${istio_tags[$key]}"
+    done
+}
+
+# Function to map our image names to upstream names
+map_our_name_to_istio() {
+    local our_name="$1"
+    
+    case "$our_name" in
+        "examples-helloworld-v1")
+            echo "examples-helloworld-v1"
+            ;;
+        "examples-helloworld-v2")
+            echo "examples-helloworld-v2"
+            ;;
+        "examples-httpbin")
+            echo "httpbin"
+            ;;
+        "examples-tcp-echo-server")
+            echo "tcp-echo-server"
+            ;;
+        *)
+            echo "$our_name"
+            ;;
+    esac
```

Comment:
This can be better handled with an associative array (you can declare it at the top to be prominent).
As an example, you can see how Shipyard does it to translates kube versions to image hashes: https://github.com/submariner-io/shipyard/blob/devel/scripts/shared/lib/clusters_kind#L8

Then the translation is a simple lookup, or return the value if not found.